csrc/fusion.cpp

```diff
@@ -26,21 +26,25 @@
 
 namespace nvfuser {
 
-static thread_local Fusion* ACTIVE_FUSION = nullptr; // NOLINT
+/*static*/ thread_local Fusion* FusionGuard::active_fusion_ = nullptr;
 
-FusionGuard::FusionGuard(Fusion* fusion) : prev_fusion{ACTIVE_FUSION} {
-  ACTIVE_FUSION = fusion;
+FusionGuard::FusionGuard(const Fusion* fusion) : prev_fusion_(active_fusion_) {
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+  active_fusion_ = const_cast<Fusion*>(fusion);
 }
 
 FusionGuard::~FusionGuard() {
-  ACTIVE_FUSION = prev_fusion;
+  active_fusion_ = prev_fusion_;
 }
 
+// Cast to non-cast because many users need it.
 Fusion* FusionGuard::getCurFusion() {
-  return ACTIVE_FUSION;
+  return active_fusion_;
 }
-void FusionGuard::setCurFusion(Fusion* fusion) {
-  ACTIVE_FUSION = fusion;
+
+void FusionGuard::setCurFusion(const Fusion* fusion) {
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+  active_fusion_ = const_cast<Fusion*>(fusion);
```

Comment:
So, do we need to support this signature? If we do need to set current fusion as a const pointer, we could likely just modify the calling code. Worst case we could put the `const_cast` _there_ instead which would be cleaner than casting away const down here.

Reply:
That's fair -- fixed.